File: src/medium/reverse-integer.ts
export const reverseInteger = (x: number): number => {
    let num = 0;
    while (x != 0) {
        num = num * 10 + (x % 10);
        x = Math.trunc(x / 10);
    }
    if (num > 0x7fffffff || num < -0x80000000) return 0;
    return num;
};

File: lib/medium/reverse_integer.dart
int reverseInteger(int x) {
  int result = 0;
  int y = x.abs();
  int s = 1;
  if (x < 0) s *= -1;
  while (y != 0) {
    result = result * 10 + (y % 10);
    y ~/= 10;
  }

  if (result > 0x7FFFFFFF || result < -0x80000000) {
    return 0;
  }
  return result * s;
}
